planner/core/fragment.go

```diff
@@ -79,29 +81,48 @@ type tasksAndFrags struct {
 }
 
 type mppTaskGenerator struct {
-	ctx     sessionctx.Context
-	startTS uint64
-	is      infoschema.InfoSchema
-	frags   []*Fragment
-	cache   map[int]tasksAndFrags
+	ctx          sessionctx.Context
+	startTS      uint64
+	localQueryID uint64
+	queryTS      uint64
+	is           infoschema.InfoSchema
+	frags        []*Fragment
+	cache        map[int]tasksAndFrags
 }
 
 // GenerateRootMPPTasks generate all mpp tasks and return root ones.
-func GenerateRootMPPTasks(ctx sessionctx.Context, startTs uint64, sender *PhysicalExchangeSender, is infoschema.InfoSchema) ([]*Fragment, error) {
+func GenerateRootMPPTasks(ctx sessionctx.Context, startTs uint64, queryTs uint64, localQueryID uint64, sender *PhysicalExchangeSender, is infoschema.InfoSchema) ([]*Fragment, error) {
 	g := &mppTaskGenerator{
-		ctx:     ctx,
-		startTS: startTs,
-		is:      is,
-		cache:   make(map[int]tasksAndFrags),
+		ctx:          ctx,
+		startTS:      startTs,
+		queryTS:      queryTs,
+		localQueryID: localQueryID,
+		is:           is,
+		cache:        make(map[int]tasksAndFrags),
 	}
 	return g.generateMPPTasks(sender)
 }
 
+var mppTaskID int64 = 1
```

Comment:
`mppTaskID` is a global var and keep increasing?

Reply:
Yes, just keep increasing is enough.